pkg/template/template_test.go

```diff
@@ -177,6 +177,59 @@ func TestParameterGenerators(t *testing.T) {
 	}
 }
 
+func TestProcessValue(t *testing.T) {
+	var template api.Template
+	if err := runtime.DecodeInto(kapi.Codecs.UniversalDecoder(), []byte(`{
+		"kind":"Template", "apiVersion":"v1",
+		"objects": [
+			{
+				"kind": "Service", "apiVersion": "v${VALUE}",
+				"metadata": {
+					"labels": {
+						"key1": "${VALUE}",
+						"key2": "$${VALUE}",
+						"s1_s1": "${STRING_1}_${STRING_1}",
+						"s1_s2": "${STRING_1}_${STRING_2}",
+						"i1": "${{INT_1}}",
```

Comment:
Create test conditions for the "error" case is described above.  Also bad json substitution error case.  I'd like a test that covers all the "broken" cases as well as the working cases.